test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc

```diff
@@ -345,6 +345,12 @@ TEST_P(ProxyProtocolTest, ErrorRecv_2) {
   EXPECT_CALL(os_sys_calls, close(_)).Times(AnyNumber()).WillRepeatedly(Invoke([this](os_fd_t fd) {
     return os_sys_calls_actual_.close(fd);
   }));
+  EXPECT_CALL(os_sys_calls, accept(_, _, _))
+      .Times(AnyNumber())
+      .WillRepeatedly(Invoke(
```

Comment:
I think this is an unfortunate use of mocking to provide real syscall behavior, but it's consistent with the existing code which basically establishes this pattern, so I think it's OK for this PR. Maybe add some TODOs above each of these blocks to clean this up in the future, this is really bad test code.

Reply:
Sure makes sense. Do you prefer a TODOs above the code blocks or an issue on github?

Reply:
Also as an observation I have seen this pattern before in many tests.

Reply:
Added a TODO in code for this in 8590fee058d197a3888261c1a541e2fd449d2c17. Let me know if I should also open an issue.